overlord/snapstate/handlers.go

```diff
@@ -1926,21 +1928,39 @@ func setMigrationFlagsinState(snapst *SnapState, snapsup *SnapSetup) {
 	}
 }
 
+// MaybeReboot will schedule a reboot if allowed by the type of system.
+func MaybeReboot(t *state.Task) error {
```

Comment:
RestartSystem from here should not be used anymore, so it should be probably be unexported and its logic merged into a MaybeRestartSystem (instead of MaybeReboot) which should live in snapstate.go as MaybeRestartSystem then?  and the core logic (dealing with boot-id etc) of it should move into overlord/restart.MaybeRestartSystem.

Reply:
Done now